src/sentry/explore/translation/discover_translation.py

```diff
@@ -0,0 +1,226 @@
+import re
+from typing import Any
+
+from sentry.discover.arithmetic import is_equation
+from sentry.discover.models import DiscoverSavedQuery
+from sentry.discover.translation.mep_to_eap import QueryParts, translate_mep_to_eap
+from sentry.explore.models import ExploreSavedQuery, ExploreSavedQueryDataset
+from sentry.integrations.slack.unfurl.discover import is_aggregate
+
+# we're going to keep the chart types from discover
+# bar = 0, line = 1, area = 2
+CHART_TYPES = {
+    "default": 2,
+    "previous": 2,
+    "top5": 2,
+    "daily": 0,
+    "dailytop5": 0,
+    "bar": 0,
+}
+
+
+def strip_negative_from_orderby(orderby):
+    """
+    This function is used to strip the negative from an orderby item.
+    """
+    if orderby.startswith("-"):
+        return orderby[1:], True
+    return orderby, False
+
+
+def _get_translated_orderby_item(orderby, columns, is_negated):
+    """
+    This function is used to translate the function underscore notation for orderby items
+    to regular function notation. We do this by stripping both the orderby item and the given columns
+    (which could be functions and fields) and then checking if it matches up to any of those stripped columns.
+    """
+    columns_stripped_list = [re.sub(r"[.()_,\s]", "", column) for column in columns]
+    joined_orderby_item = re.sub(r"[.()_,\s]", "", orderby)
+    if joined_orderby_item in columns_stripped_list:
+        try:
+            field_orderby_index = columns_stripped_list.index(joined_orderby_item)
+            converted_orderby_item = columns[field_orderby_index]
+            if is_negated:
+                converted_orderby_item = f"-{converted_orderby_item}"
+            return converted_orderby_item
+        except ValueError:
+            # if the orderby item is not in the columns, it should be dropped anyways
+            return None
+    # if the orderby item is not a field it should be dropped anyways
+    else:
+        return None
+
+
+def _format_orderby_for_translation(orderby, columns):
+    orderby_converted_list = []
+    if type(orderby) is list:
```

Comment:
to improve readability/de-dupe some of this code, you could instead check to see if orderby was string and if it was, make it a list with one element and just do the loop section of your code